src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala

```diff
@@ -0,0 +1,60 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.util.ByteString
+import io.iohk.ethereum.jsonrpc.JsonRpcController.{JsonDecoder, JsonEncoder}
+import io.iohk.ethereum.jsonrpc.Web3Service.{ClientVersionRequest, ClientVersionResponse, Sha3Request, Sha3Response}
+import org.json4s.{DefaultFormats, Formats, JValue}
+import org.json4s.JsonAST.{JArray, JString}
+import org.json4s.JsonDSL._
+import org.spongycastle.util.encoders.Hex
+
+import scala.util.{Failure, Success, Try}
+
+object JsonMethodsImplicits {
+
+  import JsonRpcErrors._
+
+  implicit val formats: Formats = DefaultFormats
+
+  implicit val web3_sha3 = new JsonDecoder[Sha3Request] with JsonEncoder[Sha3Response] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, Sha3Request] =
+      params match {
+        case Some(JArray((input: JString) :: Nil)) => tryExtractUnformattedData(input).map(Sha3Request)
+        case _ => Left(InvalidParams)
+      }
+
+    override def encodeJson(t: Sha3Response): JValue = encodeAsHex(t.data)
+  }
+
+  implicit val web3_clientVersion = new JsonDecoder[ClientVersionRequest] with JsonEncoder[ClientVersionResponse] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, ClientVersionRequest] = Right(ClientVersionRequest())
+
+    override def encodeJson(t: ClientVersionResponse): JValue = t.value
+  }
+
+  private def encodeAsHex(input: ByteString): JString =
+    JString(s"0x${Hex.toHexString(input.toArray[Byte])}")
+
+  private def encodeAsHex(input: BigInt): JString =
+    JString(s"0x${Hex.toHexString(input.toByteArray)}")
+
+  private def tryExtractUnformattedData(input: JString): Either[JsonRpcError, ByteString] = {
```

Comment:
you should use BigInt :) or if you need to accept QUANTITY but as ByteString instead of BigInt you can write separate method for that.
For `unformatted data` `0x123` is not a valid input.